shared/httputil/http_error.go

```diff
@@ -0,0 +1,162 @@
+// Copyright 2025 Woodpecker Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package httputil
+
+import (
+	"context"
+	"crypto/x509"
+	"errors"
+	"fmt"
+	"net"
+	"net/url"
+	"os"
+	"strings"
+	"syscall"
+)
+
+// EnhanceHTTPError adds detailed context to HTTP errors to help with debugging.
+func EnhanceHTTPError(err error, method, endpoint string) error {
```

Comment:
The endpoint should be removed from all errors in this method as well and shouldn't be necessary in the method anymore.

```suggestion
func EnhanceHTTPError(err error, method) error {
```